pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -722,7 +723,7 @@ InstanceSelector.SelectionResult calculateRouting(BrokerRequest brokerRequest, l
       Set<String> selectedSegments = _segmentSelector.select(brokerRequest);
       int numTotalSelectedSegments = selectedSegments.size();
       if (!selectedSegments.isEmpty()) {
-        for (SegmentPruner segmentPruner : _segmentPruners) {
+        for (SegmentPruner segmentPruner : _segmentMetadataCache.getPruners()) {
```

Comment:
We should keep both pruners and metadata fetcher in the routing entry. Fetcher shouldn't have any knowledge on the pruner